WebView/app/src/androidTest/java/com/android/samples/webviewdemo/MainActivityTest.kt

```diff
@@ -0,0 +1,158 @@
+ * limitations under the License.
+ */
+
+package com.android.samples.webviewdemo
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import android.webkit.WebView
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.web.assertion.WebViewAssertions.webMatches
+import androidx.test.espresso.web.model.Atoms.castOrDie
+import androidx.test.espresso.web.model.Atoms.script
+import androidx.test.espresso.web.sugar.Web.onWebView
+import androidx.test.espresso.web.webdriver.DriverAtoms.findElement
+import androidx.test.espresso.web.webdriver.DriverAtoms.getText
+import androidx.test.espresso.web.webdriver.DriverAtoms.webClick
+import androidx.test.espresso.web.webdriver.Locator
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertTrue
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.containsString
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+/**
+ * Launch, interact, and verify conditions in an activity that has a WebView instance.
+ */
+@RunWith(AndroidJUnit4::class)
+
+
+class MainActivityTest {
+
+    val context = ApplicationProvider.getApplicationContext<Context>()
+
+    @Rule @JvmField
+    val mainActivityRule = ActivityTestRule(MainActivity::class.java)
+    fun afterActivityLaunched() {
+      // Technically we do not need to do this - MainActivity has javascript turned on.
+      // Other WebViews in your app may have javascript turned off, however since the only way
+      // to automate WebViews is through javascript, it must be enabled.
+        onWebView().forceJavascriptEnabled()
+    }
+
+    // Test to check that the drop down menu behaves as expected
+    @Test
+    fun dropDownMenu_SanFran() {
+        mainActivityRule.getActivity()
+        onWebView()
+            .withElement(findElement(Locator.ID, "location"))
+            .perform(webClick()) // Similar to perform(click())
+            .withElement(findElement(Locator.ID, "SF"))
+            .perform(webClick()) // Similar to perform(click())
+            .withElement(findElement(Locator.ID, "title"))
+            .check(webMatches(getText(), containsString("San Francisco")))
+    }
+
+    // Test for checking createJsObject
+    @Test
+    fun jsObjectIsInjectedAndContainsPostMessage() {
+    mainActivityRule.getActivity()
+    onWebView()
+        .check(
+            webMatches(
+                script("return jsObject && jsObject.postMessage != null;", castOrDie(Boolean::class.javaObjectType)),
+                `is`(true)
+            )
+        )
+    }
+
+    @Test
+    fun valueInCallback_compareValueInput_returnsTrue(){
+        mainActivityRule.getActivity()
+        // Setup
+        val jsObjName = "jsObject"
+        val allowedOriginRules = setOf<String>("https://example.com")
+        val message = "hello"
```

Comment:
nit: `expectedMessage` would be better, not to be confused with the "actual message" (the one which comes from out of the callback)

Reply:
Ok. This made me think that we could also remove the message value in `checkingThreadCallbackRunsOn()` as it has no purpose there. Because we removed `message`, I changed the value in the `postMessage()` call to be the string hello instead of `${message}`.